build head~1 and run pr-time benchmarks on it

laithsakka opened the pull request:
This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131751

laithsakka committed:
build head~1 and run benchmarks on it

[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"




[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"




[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"




[ghstack-poisoned]

laithsakka committed:
Update on "[draft] build head~1 and run pr-time benchmarks on it"





[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 




[ghstack-poisoned]

aorenste commented on .ci/pytorch/test.sh:
```diff
@@ -400,7 +400,15 @@ pr_time_benchmarks() {
-
+  git checkout HEAD~1
+
+  # we can try firt to run develop alone, and if it worked then we do not need to run clean->develop.
+  # another suggestion was to use historical stored data insetad of building Head~1
+  python setup.py clean
+  python setup.py develop
```
Maybe
```
python setup.py develop --cmake
```
which should be good enough for 99% of changes and is a lot faster than clean.

laithsakka replied:
and we fail for the remaining 1%? i will try to mimic test_forward_backward_compatibility

aorenste commented on .ci/pytorch/test.sh:
```diff
@@ -400,7 +400,15 @@ pr_time_benchmarks() {
   echo "benchmark results on current PR: "
   cat  "$TEST_REPORTS_DIR/pr_time_benchmarks_after.txt"
-
+  git checkout HEAD~1
```
Do we need to worry about the fact that we're leaving it checked out at HEAD~1?  Maybe restore it before exiting the function.

kit1980 replied:
What if there are more than 1 commit in the PR?
You should look at `test_forward_backward_compatibility` and probably do the same.

laithsakka replied:
@aorenste I do exit after i run the pr_time_benchmarks tests, so nothing after that run, no need to restore.
@kit1980 I can check and try to rebuild same way as in test_forward_backward_compatibility, maybe i can move that to  a shared function

aorenste commented on .ci/pytorch/test.sh:
```diff
@@ -400,7 +400,15 @@ pr_time_benchmarks() {
-
+  git checkout HEAD~1
+
+  # we can try firt to run develop alone, and if it worked then we do not need to run clean->develop.
```
nit: spelling

kit1980 commented:
`test_forward_backward_compatibility` in the test.sh does basically what you want - compares with the base commit.
You should probably implement yours the same way unless there are serious reasons that will not work for you.

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka commented:
check if linux.gcp.a100 have hardware counters.

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka commented:
hide build logs,

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on current PR: 
[481](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:482)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
[482](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:483)
update_hint_regression,instruction_count,28833781796
[483](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:484)


[32061](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32062)
+ echo 'benchmark results on main: '
[32062](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32063)
benchmark results on main: 
[32063](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32064)
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
[32064](https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121#step:20:32065)
update_hint_regression,instruction_count,28514131958

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

ezyang reviewed (changes requested):
No. I am seriously skeptical of this approach. There are several reasons why I do not want to do it this way:

1. It takes a long time. If I am iterating on compile time on a fixed revision, there is no reason to keep rebuilding the HEAD~ commit over and over again. But I will have to wait for that on every PR. Building PyTorch is expensive!
2. To make matters worse, you're building PyTorch on the test runner. If it's a GPU runner, you're wasting precious GPU cycles doing a CPU only build.
3. If the only signal you are getting is pairwise performance signals between HEAD~1 and HEAD, then if someone ignores this signal and lands their PR anyway, then it is completely lost to the ages. We absolutely need some alternate dashboard like setup to durably observe regressions on main when people ignore CI signals and land anyway.

I am pretty sure we need a database of some sort. If you think you're not the right person to implement a database, let's find someone else to do it. But definitely not this.

oulgen commented:
Regarding the time it takes to build, most PRs only change python files, so if we can setup it up right, the second build should be free?

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka commented:
> No. I am seriously skeptical of this approach. There are several reasons why I do not want to do it this way:
> 
> 1. It takes a long time. If I am iterating on compile time on a fixed revision, there is no reason to keep rebuilding the HEAD~ commit over and over again. But I will have to wait for that on every PR. Building PyTorch is expensive!
> 2. To make matters worse, you're building PyTorch on the test runner. If it's a GPU runner, you're wasting precious GPU cycles doing a CPU only build.
> 3. If the only signal you are getting is pairwise performance signals between HEAD~1 and HEAD, then if someone ignores this signal and lands their PR anyway, then it is completely lost to the ages. We absolutely need some alternate dashboard like setup to durably observe regressions on main when people ignore CI signals and land anyway.
> 
> I am pretty sure we need a database of some sort. If you think you're not the right person to implement a database, let's find someone else to do it. But definitely not this.


- We can always upload the result files somewhere and have a delayed signal as well, but we do need a diff time signal for sure, because its not practical to depend on a late signal and revert not is it practical to depend on someone monitering those external delayed dashboard. 
So i think step 1 we need to get the diff time signal working. Then we can think about the next step. 

- We did have the concern about rebuilding pytorch, We discussed three other alternatives :
1. store the most recent result (The result on main) in some storage and compare against it. 
2. install nightly and compare against it. 
3. do not build from scratch but rather, apply change and try to build (this could fail in such case we do not run the test). 
4. we can do this a push time test (when its landed not while iterating). 

"I am pretty sure we need a database of some sort."
if you have ideas please share

 "If you think you're not the right person to implement a database, let's find someone else to do it."
I think lets firs decide what we want to do, then we can decide who is the right person to do it.

laithsakka commented:
> We used to have a version of this (by @robieta?), maybe he has more thoughts

ahh interesting

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka commented:
Me too

laithsakka commented:
Me too

laithsakka commented:
this comment is auto generatd testing

laithsakka commented:
this comment is auto generatd testing

laithsakka commented:
this comment is auto generatd testing ,

laithsakka committed:
Update on "build head~1 and run pr-time benchmarks on it"

This PR will checkout the commit HEAD~1 and build it and then run the benchmarks on it and store the results 
in a file. Next PR will add the comparison of numbers and the regression reporting. 
check results by checking logs o the test (https://github.com/pytorch/pytorch/actions/runs/10149355968/job/28065140121)
```
benchmark results on main: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt
update_hint_regression,instruction_count,29199009373



benchmark results on current PR: 
+ cat /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt
update_hint_regression,instruction_count,28556649567

```



[ghstack-poisoned]

laithsakka commented:
Got good amount of push back from pytorch on two builds idea will pivot to something else and close this for now.
